<details><summary>Failed to submit graph for 2cebfa95cf4698bf92cec5af72684a562538a058
</summary>Message:
Failed to run command: 'pbp generate from-feedstock --output /tmp/tmpyjbhjzvm/glib-feedstock.yaml --custom-metadata-file /tmp/tmpyjbhjzvm/custom_metadata.yaml --release-name glib-feedstock-27-2cebfa95cf4698bf92cec5af72684a562538a058 glib-feedstock/recipe', with error: 'INFO:root:Found feedstock glib-feedstock at /tmp/tmpyjbhjzvm/glib-feedstock/recipe
INFO:root:Build/Test graph from glib-feedstock
Created by sbx_user1051 on 2026-01-08 09:47:06
Command line: /opt/conda/envs/webhook-listener/bin/pbp generate from-feedstock --output /tmp/tmpyjbhjzvm/glib-feedstock.yaml --custom-metadata-file /tmp/tmpyjbhjzvm/custom_metadata.yaml --release-name glib-feedstock-27-2cebfa95cf4698bf92cec5af72684a562538a058 glib-feedstock/recipe

INFO:root:Found abs.yaml in /tmp/tmpyjbhjzvm/glib-feedstock
INFO:root:Creating graph from feedstock at: glib-feedstock
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.10 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 2 unchanging vars:(456), changing vars (2): ['blas_impl', 'python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/bin/pbp", line 10, in <module>
    sys.exit(cli())
             ~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 413, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 624, in from_feedstock
    build_configs, test_configs = expand_build_matrix(feedstock.recipe_dir_path, src_channels)
                                  ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 79, in expand_build_matrix
    _targets, build_configs, test_configs = variants_for_feedstock_and_platform(
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        recipe_dir_path, platform, src_channels
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/utils/expand_variants.py", line 544, in variants_for_feedstock_and_platform
    metadata_tuples = api.render(
        parsed.recipe,
    ...<2 lines>...
        variants=None,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/api.py", line 68, in render
    return render_metadata_tuples(
        metadata_tuples,
    ...<3 lines>...
        bypass_env_check=bypass_env_check,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 1097, in render_metadata_tuples
    for od, om in meta.get_output_metadata_set(
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        permit_unsatisfiable_variants=permit_unsatisfiable_variants,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        permit_undefined_jinja=not finalize,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        bypass_env_check=bypass_env_check,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ):
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 2790, in get_output_metadata_set
    ensure_matching_hashes(conda_packages)
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 567, in ensure_matching_hashes
    raise RecipeError(
    ...<4 lines>...
    )
conda_build.exceptions.RecipeError: Mismatching hashes in recipe. Exact pins in dependencies that contribute to the hash often cause this. Can you change one or more exact pins to version bound constraints?
Involved packages were:
Mismatching package: libglib (id h8b17d9a_1); dep: libglib 2.84.4 h77a78f3_0; consumer package: glib-tools
Mismatching package: libglib (id h8b17d9a_1); dep: libglib 2.84.4 h77a78f3_0; consumer package: glib
Mismatching package: glib-tools (id h9fad118_1); dep: glib-tools 2.84.4 h8875d55_0; consumer package: libglib
Mismatching package: glib-tools (id h9fad118_1); dep: glib-tools 2.84.4 h8875d55_0; consumer package: glib
Mismatching package: glib (id h617169b_1); dep: glib 2.84.4 h5bdd934_0; consumer package: libglib
Mismatching package: glib (id h617169b_1); dep: glib 2.84.4 h5bdd934_0; consumer package: glib-tools

'
aws lambda request id: 01d91cd0-f20a-5e29-906c-7d098d4770de
[View in Grafana](https://grafana.anacondaconnect.com/explore?schemaVersion=1&panes=%7B%2286g%22:%7B%22datasource%22:%22loki-central%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7B__aws_cloudwatch_log_group%3D%5C%22%2Faws%2Flambda%2Fpbp-graph-submitter-prod%5C%22%7D%20%7C%20json%20%7C%20requestId%20%3D%20%6001d91cd0-f20a-5e29-906c-7d098d4770de%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-central%22%7D,%22editorMode%22:%22builder%22,%22direction%22:%22backward%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22logs%22:%7B%22columns%22:%7B%220%22:%22Time%22,%221%22:%22Line%22%7D,%22visualisationType%22:%22table%22,%22labelFieldName%22:%22labels%22%7D%7D%7D%7D&orgId=1)</details>

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -33,7 +33,7 @@ source:
     - patches/0007-disable-test_fork_fail.patch  # [linux]
 
 build:
-  number: 1
+  number: 0
   # Keep in sync with
   # https://gitlab.gnome.org/GNOME/glib/-/blob/2.84.2/meson.build?ref_type=tags#L2503
   skip: True  # [py<37]
@@ -96,9 +96,7 @@ outputs:
         - zlib
         - pcre2
         - libiconv
-      run_constrained:
-        # Avoid colliding with older glib builds that didn't have a libglib output
-        - glib {{ version }} *_{{ PKG_BUILDNUM }}
+
     test:
       commands:
         - test -f ${PREFIX}/lib/libglib-2.0.0{{ SHLIB_EXT }}  # [osx]
@@ -122,7 +120,6 @@ outputs:
     build:
     requirements:
       build:
-        - gobject-introspection
         - meson >=1.4.0
         - ninja-base
         - {{ stdlib('c') }}
@@ -132,9 +129,9 @@ outputs:
         - setuptools
         - gettext {{ gettext }}  # [osx]
       host:
-        - {{ pin_subpackage("libglib", exact=True) }}
+        - {{ pin_subpackage("libglib", skip_build_id=True) }}
       run:
-        - {{ pin_subpackage("libglib", exact=True) }}
+        - {{ pin_subpackage("libglib", skip_build_id=True) }}
         - gettext >=0.21.0,<1.0a0  # [osx]
     test:
       commands:
@@ -154,7 +151,6 @@ outputs:
         - {{ pin_subpackage("libglib") }}
     requirements:
       build:
-        - gobject-introspection
         - meson >=1.4.0
         - ninja-base
         - {{ stdlib('c') }}
@@ -163,14 +159,14 @@ outputs:
         - python
         - setuptools
       host:
-        - {{ pin_subpackage("libglib", exact=True) }}
-        - {{ pin_subpackage("glib-tools", exact=True) }}
+        - {{ pin_subpackage("libglib", skip_build_id=True) }}
+        - {{ pin_subpackage("glib-tools", skip_build_id=True) }}
         - gettext {{ gettext }}  # [osx]
       run:
         - python
         - gettext >=0.21.0,<1.0a0  # [osx]
-        - {{ pin_subpackage("libglib", exact=True) }}
-        - {{ pin_subpackage("glib-tools", exact=True) }}
+        - {{ pin_subpackage("libglib", skip_build_id=True) }}
+        - {{ pin_subpackage("glib-tools", skip_build_id=True) }}
     test:
       commands:
         - test -f ${PREFIX}/lib/libglib-2.0${SHLIB_EXT}  # [not win]

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -36,7 +36,7 @@ build:
   number: 0
   # Keep in sync with
   # https://gitlab.gnome.org/GNOME/glib/-/blob/2.84.2/meson.build?ref_type=tags#L2503
-  skip: True  # [py<37 or linux]
+  # skip: True  # [py<37]
 
 requirements:
   build:

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -36,7 +36,7 @@ build:
   number: 0
   # Keep in sync with
   # https://gitlab.gnome.org/GNOME/glib/-/blob/2.84.2/meson.build?ref_type=tags#L2503
-  skip: True  # [py<37]
+  skip: True  # [py<37 or linux]
 
 requirements:
   build:

